UnitySetup/UnitySetup.psm1

```diff
@@ -49,24 +49,35 @@ class UnitySetupInstance {
     [string]$Path
 
     UnitySetupInstance([string]$path) {
-
         $currentOS = Get-OperatingSystem
-        $ivyPath = switch ($currentOS) {
-            ([OperatingSystem]::Windows) { 'Editor\Data\UnityExtensions\Unity\Networking\ivy.xml' }
-            ([OperatingSystem]::Linux) { throw "UnitySetupInstance has not been implemented on the Linux platform. Contributions welcomed!"; }
-            ([OperatingSystem]::Mac) { 'Unity.app/Contents/UnityExtensions/Unity/Networking/ivy.xml' }
+
+        if ( Test-Path "$path\modules.json" ) {
+            $modules = (Get-Content "$path\modules.json" -Raw) | ConvertFrom-Json
+
+            foreach ( $module in $modules ) {
+                $match = $module.DownloadUrl -match "(\d+)\.(\d+)\.(\d+)([fpb])(\d+)" | Out-Null
+
+                if ( $null -ne $match ) {
+                    $this.Version = [UnityVersion]$match
+                    break
+                }
+            }
         }
+        else {
+            # We'll attempt to search for the version using the ivy.xml definitions for legacy editor compatibility.
+            $ivy = Get-ChildItem -Path $path -Filter ivy.xml -Recurse -ErrorAction SilentlyContinue -Force | Select-Object -First 1
 
-        $ivyPath = [io.path]::Combine("$path", $ivyPath);
-        if (!(Test-Path $ivyPath)) { throw "Path is not a Unity setup: $path"}
-        [xml]$xmlDoc = Get-Content $ivyPath
+            if ( $null -ne $ivy ) {
+                [xml]$xmlDoc = Get-Content $ivy.FullName
+                $this.Version = [UnityVersion]$xmlDoc.'ivy-module'.info.unityVersion
+            }
+        }
 
-        if ( !($xmlDoc.'ivy-module'.info.unityVersion)) {
-            throw "Unity setup ivy is missing version: $ivyPath"
+        if ( -not $this.Version ) {
+            throw "Failed to find a valid version identifier for installation at $path!";
```

Comment:
I hit this now, despite having plenty of unity versions:
```
C:\Users\User\Source\UnitySetup.powershell [dev-UnitySetupInstance ≡]> Get-UnitySetupInstance
Failed to find a valid version identifier for installation at C:\Program Files\Unity Hub!
At C:\Users\User\Source\UnitySetup.powershell\UnitySetup\UnitySetup.psm1:77 char:13
+             throw "Failed to find a valid version identifier for inst ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Failed to find ...iles\Unity Hub!:String) [], RuntimeException
    + FullyQualifiedErrorId : Failed to find a valid version identifier for installation at C:\Program Files\Unity Hub!
```

Reply:
I'm not sure how to check if this value has been set yet. Another way to do it?